File: solutions/code/3. Longest Substring Without Repeating Characters/solution2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> hashmap;
        int longest = 0, n = s.length(), begin = 0;

        for (int i = 0; i < n; i++) {
            char c = s[i];
            if (hashmap.count(c) > 0 && hashmap[c] >= begin) {
                begin = hashmap[c] + 1;
            }
            longest = max(longest, i - begin + 1);
            hashmap[c] = i;
        }

        return longest;
    }
};

File: code/0003. Longest Substring Without Repeating Characters/solution2.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character, Integer> hashmap = new HashMap<>();
        int longest = 0, n = s.length(), begin = 0;

        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            if (hashmap.containsKey(c) && hashmap.get(c) >= begin) {
                begin = hashmap.get(c) + 1;
            }
            longest = Math.max(longest, i - begin + 1);
            hashmap.put(c, i);
        }

        return longest;
    }
}